packages/daemon/DESIGN.md

```diff
@@ -0,0 +1,90 @@
+
+Endo is a user agent for running guest programs.
+These programs receive capabilities from the user in the form of references to
+potentially powerful remote objects.
+To that end, Endo exhibits a system called Eventual Send that stretches
+references to remote objects over one or more capability transfer protocols
+(CapTP).
+
+A Promise represents a result from either the past or the future, and Eventual
+Send enables programs to interact with those eventual results, even if they are
+in another worker or on the far side of a network.
+Furthermore, Eventual Send allows messages to be pipelined to these eventual
+results to overcome any latency that might otherwise accrue if the program
+waited for each intermediate result before sending a follow-up message.
+
+Much hinges on what happens when a program restarts.
+Like most programs, Endo workers start without any memory from their previous
+run.
+
+Endo applications ultimately receive all their powers from the user.
+They receive a "powerbox" object from the user they can use to petition the
```

Comment:
"petition". I would have said "ask" or "request". No change suggested.

Reply:
The best explanation of "powerbox" is at https://www.hpl.hp.com/techreports/2006/HPL-2006-116.html , which you should both read and link to. Lots of similarities here.

Reply:
"They receive a ... object from the user"

Even though I think I know what you mean, I'm having a hard time understanding this phrase. Objects are in the computers. Users are not. Users therefore do things through user interfaces. For most purposes we can leave that implicit. But when we're talking about that very user interface by which users get access to objects, we need to be clearer.